alembic/versions/bb2cd24f0500_added_uuid_column_to_phonebook_table.py

```diff
@@ -0,0 +1,160 @@
+from sqlalchemy.dialects import postgresql
+
+
+# revision identifiers, used by Alembic.
+revision = 'bb2cd24f0500'
+down_revision = '9c426a2c7e59'
+
+dird_contact_table = sa.table(
+    'dird_contact',
+    sa.column('phonebook_id'),
+    sa.column('phonebook_uuid'),
+)
+dird_phonebook_table = sa.table(
+    'dird_phonebook',
+    sa.column('id'),
+    sa.column('uuid'),
+)
+
+
+def update_contact_phonebook_uuid():
+    return dird_contact_table.update().values(
+        phonebook_uuid=sa.select([dird_phonebook_table.c.uuid]).where(
+            dird_phonebook_table.c.id == dird_contact_table.c.phonebook_id
+        )
+    )
+
+
+def update_contact_phonebook_id():
+    return dird_contact_table.update().values(
+        phonebook_id=sa.select([dird_phonebook_table.c.id]).where(
+            dird_phonebook_table.c.uuid == dird_contact_table.c.phonebook_uuid
+        )
+    )
+
+
+def sequence_exists(conn, seq_name):
+    result = conn.execute('select sequencename from pg_sequences')
+    names = list(seq for (seq,) in result)
+    return seq_name in names
+
+
+def get_highest_phonebook_id(conn) -> int | None:
+    return conn.execute(
+        dird_phonebook_table.select([sa.func.max(dird_phonebook_table.c.id)])
+    ).scalar()
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    # remove obsolete foreign key column 'phonebook_id' from dird_contact
+    op.drop_constraint(
+        'dird_contact_phonebook_id_fkey', 'dird_contact', type_='foreignkey'
+    )
+
+    # add new primary key column uuid to dird_phonebook table
+    op.add_column(
+        'dird_phonebook',
+        sa.Column(
+            'uuid',
+            postgresql.UUID(),
+            server_default=sa.text('uuid_generate_v4()'),
+            nullable=False,
+        ),
+    )
+    # replace integer primary key 'id' with uuid
+    op.drop_constraint('dird_phonebook_pkey', 'dird_phonebook', type_='primary')
+    op.create_primary_key('dird_phonebook_pkey', 'dird_phonebook', ['uuid'])
+
+    # make sure existing autoincremented id column remains functional
+    op.create_unique_constraint('dird_phonebook_id', 'dird_phonebook', ['id'])
+    if not sequence_exists(op.get_bind(), 'dird_phonebook_id_seq'):
+        sequence = (
+            sa.Sequence(
+                'dird_phonebook_id_seq',
+                start=get_highest_phonebook_id(op.get_bind()) or 1,
+            ),
+        )  # TODO: check for highest existing value
+        op.execute(sa.schema.CreateSequence(sequence))
+
+    op.alter_column(
+        'dird_phonebook',
+        'id',
+        server_default=sa.text('nextval(\'dird_phonebook_id_seq\'::regclass)'),
+        nullable=False,
+    )
+
+    # add new foreign key to dird_contact table referencing dird_phonebook uuid column
+    op.add_column(
+        'dird_contact',
+        sa.Column(
+            'phonebook_uuid',
+            postgresql.UUID(),
+            server_default=sa.text('uuid_generate_v4()'),
```

Comment:
isn't that the behavior we want?

Reply:
It's also what is done in other migrations adding a `uuid` column.

Reply:
Adding `uuid` column is fine, because we want to generate it. But this column is a foreign key, if we generate uuid, your next step (that add foreign key) will fail anyways. Maybe I miss something, please call me to talk about it if it's not clear

Reply:
Oh woops, that's a mistake indeed